File: apps/api/app/schemas/planner.py
from pydantic import BaseModel, Field
from typing import List, Optional


class TripPlanRequest(BaseModel):
    departureCity: str
    destination: str
    startDate: str
    endDate: str
    travellers: str
    budget: str
    style: str
    interests: List[str]
    notes: str = ""


class DayActivity(BaseModel):
    title: str
    description: str
    meta: Optional[str] = None


class DayPlan(BaseModel):
    title: str
    date: str
    activities: List[DayActivity]


class HotelRecommendation(BaseModel):
    name: str
    address: str
    imageQuery: str
    bookingQuery: str
    summary: str
    price: str
    tag: str
    image_url: Optional[str] = None
    meta: Optional[str] = None
    bestMatch: bool = False


class FlightRecommendation(BaseModel):
    type: str
    airline: str
    departureAirport: str
    arrivalAirport: str
    departureDate: str
    departureTime: str
    arrivalDate: str
    arrivalTime: str
    duration: str
    stops: str
    price: str
    cabinClass: str


class BudgetSummary(BaseModel):
    total: str
    trend: str
    travelTime: str
    activitiesCount: int


class RestaurantRecommendation(BaseModel):
    name: str
    cuisine: str
    rating: str
    price: str
    description: str
    address: str


class TripPlanResponse(BaseModel):
    destination: str
    latitude: float
    longitude: float
    overview: str
    budgetSummary: BudgetSummary
    hotels: List[HotelRecommendation]
    restaurants: List[RestaurantRecommendation]
    flights: List[FlightRecommendation]
    days: List[DayPlan]
    travelTips: str
    packingSuggestions: List[str]

File: apps/web/src/types/planner.ts
export interface TripPlanRequest {
  departureCity: string;
  destination: string;
  startDate: string;
  endDate: string;
  travellers: string;
  budget: string;
  style: string;
  interests: string[];
  notes?: string;
}

export interface DayActivity {
  title: string;
  description: string;
  meta?: string;
}

export interface DayPlan {
  title: string;
  date: string;
  activities: DayActivity[];
}

export interface HotelRecommendation {
  name: string;
  address: string;
  imageQuery: string;
  bookingQuery: string;
  summary: string;
  price: string;
  tag: string;
  image_url?: string;
  meta?: string;
  bestMatch?: boolean;
}

export interface FlightRecommendation {
  type: string;
  airline: string;
  departureAirport: string;
  arrivalAirport: string;
  departureDate: string;
  departureTime: string;
  arrivalDate: string;
  arrivalTime: string;
  duration: string;
  stops: string;
  price: string;
  cabinClass: string;
}

export interface BudgetSummary {
  total: string;
  trend: string;
  travelTime: string;
  activitiesCount: number;
}

export interface RestaurantRecommendation {
  name: string;
  cuisine: string;
  rating: string;
  price: string;
  description: string;
  address: string;
}

export interface TripPlanResponse {
  destination: string;
  latitude: number;
  longitude: number;
  overview: string;
  budgetSummary: BudgetSummary;
  hotels: HotelRecommendation[];
  restaurants: RestaurantRecommendation[];
  flights: FlightRecommendation[];
  days: DayPlan[];
  travelTips: string;
  packingSuggestions: string[];
}
